### PR TITLE
[PN-126] Combine spot_driver and spot_driver_cpp

### PR DESCRIPTION
**Pull request by gbrooks-bdai**
[PN-126] Move driver python code and rename but not configs and launch

[PN-126] Move configs and launch

[PN-126]: https://theaiinstitute.atlassian.net/browse/PN-126?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PN-126]: https://theaiinstitute.atlassian.net/browse/PN-126?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by gbrooks-bdai** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #241** <a href="https://app.graphite.dev/github/pr/bdaiinstitute/spot_ros2/241?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/bdaiinstitute/spot_ros2/241?utm_source=stack-comment).

File: spot_driver/package.xml [hunk 1]
```diff
@@ -3,37 +3,43 @@
...
+  <depend>python3-protobuf</depend>
+  <depend>rclcpp</depend>
+  <depend>rclcpp_components</depend>
+  <depend>rclpy</depend>
+  <depend>robot_state_publisher</depend>
+  <depend>rosidl_default_runtime</depend>
+  <depend>rviz2</depend>
+  <depend>sensor_msgs</depend>
+  <depend>spot_description</depend>
+  <depend>spot_driver_cpp</depend>
```
Comment (jschornak-bdai) [inline spot_driver/package.xml:30]: Since this package doesn't exist anymore we can delete this dependency.

**Review by jschornak-bdai — CHANGES_REQUESTED** [review]
- There are a few uses of the old `spot_driver_cpp` package name in `spot_driver/launch/spot_driver.launch.py` that'll need to be updated.
- The C++ header files in `spot_driver/include/spot_driver_cpp` should be moved to `spot_driver/include/spot_driver`. This will be fairly annoying since all the uses of these headers in the C++ code will need to be updated too, but I think it's a requirement/expectation for this directory to have a name that matches its package.

File: spot_driver/package.xml (hunk 1 above)
Comment (gbrooks-bdai) [reply spot_driver/package.xml:30]: Good point

Comment (gbrooks-bdai) [reply spot_driver/package.xml]: Done

**Comment by gbrooks-bdai** [comment]
Interesting that things are still building and tests aren't failing

**Comment by gbrooks-bdai** [comment]
oh, I see why. My bad

**Comment by gbrooks-bdai** [comment]
Ok @jschornak-bdai , can you check again?

**Review by tcappellari-bdai — COMMENTED** [review]
I tried to start the driver like I usually would but am getting errors (sounds like a launch file has some formatting error that causes some TF errors down the line or something 

`[INFO] [launch]: All log files can be found below /home/tcappellari/.ros/log/2024-02-05-15-18-51-093711-tcappellari-LT-46083
[INFO] [launch]: Default logging verbosity is set to INFO
[ERROR] [spot_ros2-1]: exception occurred while executing process:
Traceback (most recent call last):
  File "/opt/ros/humble/lib/python3.10/site-packages/launch/actions/execute_local.py", line 546, in __execute_process
    transport, self._subprocess_protocol = await async_execute_process(
  File "/opt/ros/humble/lib/python3.10/site-packages/osrf_pycommon/process_utils/async_execute_process_asyncio/impl.py", line 139, in async_execute_process
    transport, protocol = await _async_execute_process_nopty(
  File "/opt/ros/humble/lib/python3.10/site-packages/osrf_pycommon/process_utils/async_execute_process_asyncio/impl.py", line 45, in _async_execute_process_nopty
    transport, protocol = await loop.subprocess_exec(
  File "/usr/lib/python3.10/asyncio/base_events.py", line 1681, in subprocess_exec
    transport = await self._make_subprocess_transport(
  File "/usr/lib/python3.10/asyncio/unix_events.py", line 207, in _make_subprocess_transport
    transp = _UnixSubprocessTransport(self, protocol, args, shell,
  File "/usr/lib/python3.10/asyncio/base_subprocess.py", line 36, in __init__
    self._start(args=args, shell=shell, stdin=stdin, stdout=stdout,
  File "/usr/lib/python3.10/asyncio/unix_events.py", line 799, in _start
    self._proc = subprocess.Popen(
  File "/usr/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.10/subprocess.py", line 1751, in _execute_child
    self._posix_spawn(args, executable, env, restore_signals,
  File "/usr/lib/python3.10/subprocess.py", line 1696, in _posix_spawn
    self.pid = os.posix_spawn(executable, args, env, **kwargs)
OSError: [Errno 8] Exec format error: '/workspaces/bdai/_build/install/spot_driver/lib/spot_driver/spot_ros2'
`
then a bunch of nodes and stuff launch fine until finally

`[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/Fausto/register_node_back' in container '/Fausto/container'
[component_container_mt-5] [ERROR] [1707146334.064975125] [Fausto.register_node_back]: TF2 exception:
[component_container_mt-5] "Fausto/back_fisheye" passed to lookupTransform argument target_frame does not exist. 
[component_container_mt-5] [ERROR] [1707146334.064976377] [Fausto.register_node_frontleft]: TF2 exception:
[component_container_mt-5] "Fausto/frontleft_fisheye" passed to lookupTransform argument target_frame does not exist. 
[component_container_mt-5] [ERROR] [1707146334.065850051] [Fausto.register_node_frontright]: TF2 exception:
[component_container_mt-5] "Fausto/frontright_fisheye" passed to lookupTransform argument target_frame does not exist. 
[component_container_mt-5] [ERROR] [1707146334.066356712] [Fausto.register_node_left]: TF2 exception:
[component_container_mt-5] "Fausto/left_fisheye" passed to lookupTransform argument target_frame does not exist. 
[component_container_mt-5] [ERROR] [1707146334.067264030] [Fausto.register_node_right]: TF2 exception:
[component_container_mt-5] "Fausto/right_fisheye" passed to lookupTransform argument target_frame does not exist. `

**Review by tcappellari-bdai — APPROVED** [review]
lgtm

**Comment by gbrooks-bdai** [comment]
TIL: https://roboticsbackend.com/ros2-package-for-both-python-and-cpp-nodes/#:~:text=Important%3A%20you%20have%20to%20add%20a%20shebang%20line%20first%20thing%20in%20the%20py_node.py%20file%3A

File: spot_driver/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# spot_driver_cpp
```
Comment (jschornak-bdai) [inline spot_driver/README.md:1]: Update the description in this README to reflect that the package contains both C++ and Python code

**Review by jschornak-bdai — COMMENTED** [review]
One very minor little thing and then I think this is all set:

**Review by jschornak-bdai — APPROVED** [review]

**Comment by gbrooks-bdai** [comment]
### Merge activity

* **Feb 6, 11:24 AM EST**: @gbrooks-bdai merged this pull request with [Graphite](https://app.graphite.dev/github/pr/bdaiinstitute/spot_ros2/241).

**Pull request merged**